### PR TITLE
Added CRUD functionality

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
-  I've just added CRUD functionality for notes + color-picker component.

#### Pages

- [x] [pages/NotesFeed.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-92ec52f2623e6cb2fed16a02cbcc9715c93d714b792a6dd4cef70fd883991f92)

#### Component files

- [x] [components/AddEditNote.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-1a3e8710657eca4b2347099df01302643bbea81aab3c53f5b673442a17361000)
- [x] [components/Note.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-ca4ff88786f04ecde5a7c6f0ad9692a34e8aa9a48912291c9fefaded8d9b082e)
- [x] [components/Sidebar.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-b2415db1fb43f1726a88845570f8db1a75ae284c1b612163c72280c92761e765)
- [x] [components/Color.js](https://github.com/nayyyhaa/NotesFeed/commit/bdf534f4f130504524f07ffa45d03c2c24b77d48#diff-5a9d68bb241547cdcc8301f7f49bc719f4224580b6fb10b533ac9b7c4a66aa42)
- [x] [components/ColorPickerContainer.js](https://github.com/nayyyhaa/NotesFeed/commit/bdf534f4f130504524f07ffa45d03c2c24b77d48#diff-aab2f3088ac236564fb8526864b7f964a7b6eaaafe110135b793e91c8c28b103)

 #### Contexts  + Reducers

- [x] [contexts/ModelContext.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-cda7d7cb15b9fb68100bf3d20b66c0bdf38b9f65c5f1ff4a2ba3606ddb732565)
- [x] [contexts/NoteContext.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-8e8fea03c7d6522afce2ee3fca3792ef83ecd4d5980168f2f0cbfcf9aaf3c43b)
- [x] [reducers/noteReducer.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-2971681933b22d59e0e7b96416e9e6c3f6fe70f012dde415ebc6a6ac5ec431e0)

#### utils:
- [x] [utils/formatDate.js](https://github.com/nayyyhaa/NotesFeed/commit/2041fef28d8373a1d4988b57ecbd17b8ea5a87e6#diff-a5b18423be5833e4b09575e7ef254f9274d677ee47fa2f5b3f313e3acbf5a34e)

#### files to *ignore*
- [ ] package.json files
- [ ] files with just few import changed
- [ ] CSS files

### Live preview : https://deploy-preview-2--notesfeed.netlify.app/

### CODE REVIEW

File: src/components/AddEditNote.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import { faTimes, faPalette, faTag } from "@fortawesome/free-solid-svg-icons";
+import { useNote } from "contexts/NoteContext";
+import { useModal } from "contexts/ModelContext";
+import { ColorPickerContainer } from "./ColorPickerContainer";
+
+export const AddEditNote = () => {
+  const { modalData, showModal, setModalClose } = useModal();
+  const [checkFormValidity, setFormValid] = useState(false);
+  const [isColorPickerOpen, setIsColorPicker] = useState(false);
+  const { editmodeon, note } = modalData;
```
Comment (yash9263) [inline src/components/AddEditNote.js]: camel case for variable names

**Review by MayankKumar10 — APPROVED** [review]
Everything looks good to me you can merge it.

**Review by yash9263 — APPROVED** [review]

**Comment by yash9263** [comment]
Everything looks good. The code is clean and modular. UI looks clean and beautiful.

File: src/components/AddEditNote.js (hunk 1 above)
Comment (nayyyhaa) [reply src/components/AddEditNote.js]: Done with the changes.

**Comment by nayyyhaa** [comment]
Thanks @MayankKumar10 & @yash9263 for reviewing. Merging the PR.

**Pull request merged**